### PR TITLE
feat: Add HashiCorp HCP Consul and Envoy

### PR DESCRIPTION
**Pull request by msummers-nr**
# Summary

HashiCorp HCP Consul and Envoy Quick Starts, including detailed Terraform and kubectl installation directions for the HashiCups demo application running under EKS

## Pre merge checklist

- [ N/A] Did you check you NRQL syntax? - Not applicable
- [ N/A] Did you check your dashboard image quality? -  Not applicable
- [ N/A] Did you check that your alerts actually work?- Not applicable
- [ X] Did you include an InstallPlan and Documentation reference?
- [ X] Did you check your descriptive content for voice, tone, spelling and grammar errors?
- [ N/A] Did you attach images of your dashboards to the PR so we can see them working?- Not applicable

### Screenshots

Not applicable

### CODE REVIEW

**Comment by aswanson-nr** [comment]
Hey @msummers-nr, 
Thanks so much for this contribution! I wanted to chat about some of these other config files added in this pull request. We mainly just keep files relevant to the definition of the quickstart in the repository and link out to instrumentation instructions, that way the information is available to users that don't install via the quickstart. Could you move config files and the readme to a newrelic-experimental repository and then link out to it? cc @zstix

File: quickstarts/hashicorp/hcp-consul/config.yml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+  - hcp-consul
+  - hashicorp
+  - consul
+  - nrlabs
+  - nrlabs-data
+
+# Documentation references
+documentation:
+  - name: Configuration documentation
+    url: https://github.com/newrelic-experimental/hashicorp-quickstart-annex/tree/main/hcp-consul/README.md
```
Comment (aswanson-nr) [inline quickstarts/hashicorp/hcp-consul/config.yml:35]: I'm getting a 404 from this link

**Review by aswanson-nr — COMMENTED** [review]
Hey @msummers-nr! Thanks for much for making those changes!
Both of the links you added are currently 404ing, could the repository be private?

File: quickstarts/hashicorp/hcp-envoy/config.yml [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  - hashicorp
+  - envoy
+  - envoy-proxy
+  - nrlabs
+  - nrlabs-data
+
+# Documentation references
+documentation:
+  - name: Configuration documentation
+    url: https://github.com/newrelic-experimental/hashicorp-quickstart-annex/tree/main/hcp-envoy/README.md
```
[the remark this answers is not in the document]
Comment (msummers-nr) [reply quickstarts/hashicorp/hcp-envoy/config.yml:36]: @aswanson-nr Fixed. My apologies.

File: quickstarts/hashicorp/hcp-consul/config.yml (hunk 1 above)
Comment (msummers-nr) [reply quickstarts/hashicorp/hcp-consul/config.yml:35]: @aswanson-nr Repo is fixed, sorry about that.

File: quickstarts/hashicorp/hcp-consul/config.yml [hunk 3]
```diff
@@ -0,0 +1,39 @@
+# Name of the quickstart (required)
+# sets the URL name of the quickstart on public I/O
+name: hcp-consul
```
Comment (aswanson-nr) [inline quickstarts/hashicorp/hcp-consul/config.yml:3]:
```suggestion
slug: hcp-consul
```

**Comment by aswanson-nr** [comment]
Hey @msummers-nr! We're having some issues with the validation workflows, but we'll get this merged ASAP.

**Review by aswanson-nr — APPROVED** [review]

**Pull request merged**